sql/timeseries/cruxPassesCWV.sql

```diff
@@ -0,0 +1,61 @@
+#standardSQL
+# Passes Core Web Vitals by device
+
+CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+WITH
+  base AS (
+  SELECT
+    yyyymm,
+    origin,
+    device,
+
+    fast_fid,
+    avg_fid,
+    slow_fid,
+
+    fast_lcp,
+    avg_lcp,
+    slow_lcp,
+
+    small_cls,
+    medium_cls,
+    large_cls
+
+  FROM
+    `chrome-ux-report.materialized.device_summary`
+  WHERE
+    device IN ('desktop','phone')
+    AND yyyymm >= 201909
+  )
+
+SELECT
+  REGEXP_REPLACE(CAST(yyyymm AS STRING), '(\\d{4})(\\d{2})', '\\1_\\2_01') AS date,
+  UNIX_DATE(CAST(REGEXP_REPLACE(CAST(yyyymm AS STRING), '(\\d{4})(\\d{2})', '\\1-\\2-01') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
+  IF(device = 'desktop', 'desktop', 'mobile') AS client,
+  SAFE_DIVIDE(
+      COUNT(DISTINCT IF(
+          IS_GOOD(fast_fid, avg_fid, slow_fid) AND
+          IS_GOOD(fast_lcp, avg_lcp, slow_lcp) AND
+          IS_GOOD(small_cls, medium_cls, large_cls), origin, NULL)),
+      COUNT(DISTINCT IF(
+          IS_NON_ZERO(fast_fid, avg_fid, slow_fid) AND
+          IS_NON_ZERO(fast_lcp, avg_lcp, slow_lcp) AND
+          IS_NON_ZERO(small_cls, medium_cls, large_cls), origin, NULL))) AS percent,
```

Comment:
One thing that needs to be clarified for CWV in general (and should be reflected here) is that many sites (~30% of them) that are lacking FID data should still be assessed at the CWV level, rather than the current behavior of requiring all 3 metrics in order to be assessed. I'll need to think more through how to best capture that in the query, but if you have an idea, go ahead and implement it and I'd be happy to review.

Reply:
Question is, should we change the fastFID query (and fastLCP and fastCLS) in a similar way to count null as fast? And should it be based on any CWV existing?

If I change the fastFID query to this:

```sql
#standardSQL
# Fast FID by device

CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
  good / (good + needs_improvement + poor) >= 0.75
);

CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
  good + needs_improvement + poor > 0
);

SELECT
  REGEXP_REPLACE(CAST(yyyymm AS STRING), '(\\d{4})(\\d{2})', '\\1_\\2_01') AS date,
  UNIX_DATE(CAST(REGEXP_REPLACE(CAST(yyyymm AS STRING), '(\\d{4})(\\d{2})', '\\1-\\2-01') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
  IF(device = 'desktop', 'desktop', 'mobile') AS client,
  SAFE_DIVIDE(
      COUNT(DISTINCT IF(
          (p75_fid IS NULL OR IS_GOOD(fast_fid, avg_fid, slow_fid)), origin, NULL)), 
      COUNT(DISTINCT origin)) * 100 AS percent
FROM
  `chrome-ux-report.materialized.device_summary`
WHERE
  device IN ('desktop','phone')
  AND yyyymm >= 201806
  AND (p75_fid IS NOT NULL OR p75_lcp IS NOT NULL OR p75_cls IS NOT NULL)
GROUP BY
  date,
  timestamp,
  client
ORDER BY
  date DESC,
  client
```

I see a 6% improvement for mobile for fast FID:

Date | desktop_old | desktop_new | desktop_difference | mobile_old | mobile_new | mobile_diff
-- | -- | -- | -- | -- | -- | --
2021_04_01 | 99.91% | 99.93% | 0.02% | 82.52% | 88.11% | 5.59%
2021_03_01 | 99.90% | 99.93% | 0.03% | 82.11% | 87.94% | 5.82%
2021_02_01 | 99.89% | 99.92% | 0.03% | 81.53% | 87.79% | 6.26%
2021_01_01 | 99.89% | 99.92% | 0.03% | 81.36% | 87.76% | 6.40%
2020_12_01 | 99.87% | 99.91% | 0.04% | 81.14% | 87.67% | 6.54%
2020_11_01 | 99.86% | 99.90% | 0.04% | 80.51% | 87.19% | 6.68%
2020_10_01 | 99.84% | 99.88% | 0.05% | 80.09% | 87.00% | 6.92%
2020_09_01 | 99.81% | 99.86% | 0.05% | 79.62% | 86.67% | 7.05%
2020_08_01 | 99.79% | 99.84% | 0.06% | 79.58% | 86.68% | 7.10%
2020_07_01 | 99.78% | 99.84% | 0.06% | 79.48% | 86.47% | 6.99%
2020_06_01 | 99.75% | 99.82% | 0.07% | 78.80% | 86.01% | 7.21%
2020_05_01 | 99.62% | 99.75% | 0.13% | 78.15% | 85.88% | 7.73%
2020_04_01 | 99.60% | 99.73% | 0.14% | 77.89% | 85.75% | 7.86%
2020_03_01 | 99.65% | 99.77% | 0.12% | 77.17% | 85.28% | 8.11%
2020_02_01 | 99.65% | 99.77% | 0.12% | 77.13% | 85.27% | 8.14%
2020_01_01 | 99.65% | 99.77% | 0.11% | 78.26% | 85.97% | 7.71%
2019_12_01 | 99.66% | 99.78% | 0.11% | 77.96% | 85.91% | 7.95%
2019_11_01 | 99.71% | 99.81% | 0.10% | 77.30% | 85.39% | 8.09%
2019_10_01 | 99.70% | 99.80% | 0.10% | 76.85% | 84.90% | 8.05%
2019_09_01 | 99.70% | 99.71% | 0.01% | 76.96% | 79.36% | 2.40%
2019_08_01 | 99.73% | 99.83% | 0.10% | 76.67% | 85.61% | 8.94%
2019_07_01 | 99.67% | 99.89% | 0.22% | 76.97% | 92.63% | 15.66%
2019_06_01 | 99.67% | 99.79% | 0.12% | 76.15% | 85.43% | 9.29%
2019_05_01 | 99.67% | 99.81% | 0.14% | 75.79% | 86.47% | 10.67%
2019_04_01 | 99.42% | 99.42% | 0.00% | 82.32% | 82.32% | 0.00%
2019_03_01 | 99.23% | 99.23% | 0.00% | 79.89% | 79.89% | 0.00%
2019_02_01 | 99.21% | 99.21% | 0.00% | 79.64% | 79.64% | 0.00%
2019_01_01 | 99.22% | 99.22% | 0.00% | 79.26% | 79.26% | 0.00%
2018_12_01 | 99.24% | 99.24% | 0.00% | 78.35% | 78.35% | 0.00%
2018_11_01 | 99.29% | 99.29% | 0.00% | 76.78% | 76.78% | 0.00%
2018_10_01 | 99.22% | 99.22% | 0.00% | 75.98% | 75.98% | 0.00%
2018_09_01 | 98.98% | 98.98% | 0.00% | 75.73% | 75.73% | 0.00%
2018_08_01 | 98.79% | 98.79% | 0.00% | 76.40% | 76.40% | 0.00%
2018_07_01 | 98.79% | 98.79% | 0.00% | 76.32% | 76.32% | 0.00%
2018_06_01 | 98.79% | 98.79% | 0.00% | 76.15% | 76.15% | 0.00%

Reply:
FID is the only metric that should be ignored if null when assessing whether the origin has good CWV overall. The other metrics are required.

When assessing whether FID specifically was fast for an origin and it has no FID data, fastFID should evaluate to null.